### PR TITLE
Remove deprecation warnings introduced by Ocean 4

### LINKED ISSUE #19: example still uses bqm.iter_variables()
When I run this I get the following deprecation message: 

```
UserWarning: bqm.iter_variables() is deprecated since dimod 0.10.0, use iter(bqm.variables) instead.
  for i in bqm.iter_variables())
/workspace/mutual-information-feature-selection/titanic.py:93: DeprecationWarning: bqm.iter_neighbors(v) is deprecated since dimod 0.10.0, use (v for v, _ in bqm.iter_neighborhood(v)) instead.
```

Can we update example to use Ocean 4?

### PR DESCRIPTION
**Pull request by hhtong**
Fixes #19

### CODE REVIEW

File: titanic.py [hunk 1]
```diff
@@ -91,8 +91,8 @@ def maximum_energy_delta(bqm):
     """Compute conservative bound on maximum change in energy when flipping a single variable"""
     return max(abs(bqm.get_linear(i))
                + sum(abs(bqm.get_quadratic(i,j))
-                     for j in bqm.iter_neighbors(i))
-               for i in bqm.iter_variables())
+                     for j, _ in bqm.iter_neighborhood(i))
+               for i in iter(bqm.variables))
```
Comment (randomir) [inline titanic.py]: This can just be:
```suggestion
               for i in bqm.variables)
```

File: titanic.py [hunk 2]
```diff
@@ -91,8 +91,8 @@ def maximum_energy_delta(bqm):
     """Compute conservative bound on maximum change in energy when flipping a single variable"""
     return max(abs(bqm.get_linear(i))
                + sum(abs(bqm.get_quadratic(i,j))
-                     for j in bqm.iter_neighbors(i))
-               for i in bqm.iter_variables())
+                     for j, _ in bqm.iter_neighborhood(i))
```
Comment (randomir) [inline titanic.py:94]: This doesn't work in `dimod~=0.9.0` (Ocean < 4).

I don't think there's a method for iterating neighbors that exists in both dimod 0.9 and 0.10, but @arcondello should confirm.

If that's so, then we have a choice:
- lower-bound examples to Ocean 4
- tolerate deprecation messages
- de-deprecate `iter_neighbors`/`iter_quadratic` and/or rename `iter_neighborhood` to `iter_neighbors`.

I'm not sure why we deprecated `iter_quadratic`.

Comment (hhtong) [reply titanic.py:94]: This is the only example that used `iter_neighbors`, so I would vote for raising the lower bound (if there isn't a method that works in both 0.9 and 0.10)

File: titanic.py (hunk 1 above)
Comment (hhtong) [reply titanic.py]: 😅  thanks!

File: titanic.py (hunk 2 above)
Comment (randomir) [reply titanic.py:94]: That makes sense, @hhtong, let's raise the lower bound for this example.

Comment (hhtong) [reply titanic.py:94]: Done

**Comment by randomir** [comment]
Does it need a rebase to fix failing tests?

**Comment by hhtong** [comment]
> Does it need a rebase to fix failing tests?

I don't think so, the tests are failing because of no API token

**Pull request merged**